dspace-api/src/main/java/org/dspace/statistics/AnonymizeStatistics.java

```diff
@@ -0,0 +1,273 @@
+        TIME_LIMIT = format(calendar, DATE_FORMAT_8601);
+    }
+
+    private AnonymizeStatistics() {
+
+    }
+
+
+    public static void main(String... args) throws ParseException {
+
+        parseCommandLineOptions(createCommandLineOptions(), args);
+        anonymizeStatistics();
+    }
+
+    private static Options createCommandLineOptions() {
+
+        Options options = new Options();
+        options.addOption(
+                builder(HELP_OPTION)
+                        .longOpt("help")
+                        .desc("Print the usage of the script")
+                        .hasArg(false)
+                        .build()
+        );
+
+        options.addOption(
+                builder(SLEEP_OPTION)
+                        .longOpt("sleep")
+                        .desc("Sleep a certain time given in milliseconds between each solr request")
+                        .hasArg(true)
+                        .build()
+        );
+
+        options.addOption(
+            builder(BATCH_SIZE_OPTION)
+                .longOpt("batch")
+                .desc("The amount of Solr records to be processed per batch (defaults to 100)")
+                .hasArg(true)
+                .build()
+        );
+
+        options.addOption(
+            builder(THREADS_OPTION)
+                .longOpt("threads")
+                .desc("The amount of threads used by the script (defaults to 2")
+                .hasArg(true)
+                .build()
+        );
+
+        return options;
+    }
+
+    private static void parseCommandLineOptions(Options options, String... args) throws ParseException {
+
+        CommandLine commandLine = new DefaultParser().parse(options, args);
+
+        if (commandLine.hasOption(HELP_OPTION)) {
+            printHelp(options);
+            System.exit(-1);
+        }
+
+        if (commandLine.hasOption(SLEEP_OPTION)) {
+            sleep = parseInt(commandLine.getOptionValue(SLEEP_OPTION));
+        }
+
+        if (commandLine.hasOption(BATCH_SIZE_OPTION)) {
+            batchSize = parseInt(commandLine.getOptionValue(BATCH_SIZE_OPTION));
+        }
+
+        if (commandLine.hasOption(THREADS_OPTION)) {
+            threads = parseInt(commandLine.getOptionValue(THREADS_OPTION));
+        }
+    }
+
+    private static void printHelp(Options options) {
+        new HelpFormatter().printHelp("dsrun " + AnonymizeStatistics.class.getCanonicalName(), options);
+    }
+
+    private static void printInfo(String info) {
+        System.out.println(info);
+        log.info(getHeader(context, action, info));
+    }
+
+    private static void printWarning(String warning) {
+        System.out.println(warning);
+        log.warn(getHeader(context, action, warning));
+    }
+
+    private static void printError(Exception error) {
+        error.printStackTrace();
+        log.error(getHeader(context, action, error.getMessage()), error);
+    }
+
+
+    private static void anonymizeStatistics() {
```

Comment:
Even though this is a private method, it contains most of the logic of the class. So, it should have JavaDocs or inline comments to describe *how* it is anonymizing statistics, etc.